src/simsopt/mhd/bootstrap.py

```diff
@@ -0,0 +1,740 @@
+                            index_of_min,
+                            bounds=((0, ntheta), (0, nphi)))
+            modBmin = soln.fun
+            soln = minimize(lambda x: -np.ravel(modB_spline(x[0], x[1])),
+                            index_of_max,
+                            bounds=((0, ntheta), (0, nphi)))
+            modBmax = -soln.fun
+            Bmin[js] = modBmin
+            Bmax[js] = modBmax
+            w = modBmax / modBmin
+            epsilon[js] = (w - 1) / (w + 1)
+
+            def integrand(lambd):
+                # This function gives lambda / <sqrt(1 - lambda B)>:
+                return lambd / (np.mean(np.sqrt(1 - lambd * modB[:, :, js]) * sqrtg[:, :, js]) \
+                                / (fourpisq * dVds[js]))
+
+            integral = quad(integrand, 0, 1 / modBmax)
+            f_t[js] = 1 - 0.75 * fsa_B2[js] * integral[0]
+
+    elif modB.ndim == 2:
+        # Input arrays are 2D, with no phi dependence.
+
+        twopi = 2 * np.pi
+        dVds = np.mean(sqrtg, axis=0) / twopi
+        fsa_B2 = np.mean(modB * modB * sqrtg, axis=0) / (twopi * dVds)
+        fsa_1overB = np.mean(sqrtg / modB, axis=0) / (twopi * dVds)
+
+        # Make a slightly enlarged version of the input array with the
+        # first row and column appended at the ends, for periodicity.
+        modB_big = np.zeros((ntheta + 1, ns))
+        modB_big[:ntheta, :] = modB
+        modB_big[-1, :] = modB[0, :]
+
+        theta = np.arange(ntheta + 1)
+        for js in range(ns):
+            index_of_min = np.argmin(modB_big[:, js])
+            index_of_max = np.argmax(modB_big[:, js])
+            modB_spline = interp1d(theta, modB_big[:, js], kind='cubic')
+            bounds = Bounds(0, ntheta)
+            soln = minimize(modB_spline,
+                            [index_of_min],
+                            bounds=bounds)
+            modBmin = soln.fun
+            soln = minimize(lambda x: -modB_spline(x[0]),
+                            [index_of_max],
+                            bounds=bounds)
+            modBmax = -soln.fun
+            Bmin[js] = modBmin
+            Bmax[js] = modBmax
+            w = modBmax / modBmin
+            epsilon[js] = (w - 1) / (w + 1)
+
+            def integrand(lambd):
+                # This function gives lambda / <sqrt(1 - lambda B)>:
+                return lambd / (np.mean(np.sqrt(1 - lambd * modB[:, js]) * sqrtg[:, js]) \
+                                / (twopi * dVds[js]))
+
+            integral = quad(integrand, 0, 1 / modBmax)
+            f_t[js] = 1 - 0.75 * fsa_B2[js] * integral[0]
+
+    else:
+        raise ValueError('Input arrays must be 2D or 3D')
+
+    logging.debug(f'Bmin: {Bmin}  Bmax: {Bmax}  epsilon: {epsilon}  '
+                  f'fsa_B2: {fsa_B2}  fsa_1overB: {fsa_1overB}  f_t: {f_t}')
+    return Bmin, Bmax, epsilon, fsa_B2, fsa_1overB, f_t
+
+
+def j_dot_B_Redl_fits(s, ne, Te, Ti, Zeff, G, R, iota, epsilon, f_t, psi_edge, helicity_N):
```

Comment:
Is there a reason for the name j_dot_B_Redl_fits? Since j_dot_B_Redl is currently serving as a helper function for j_dot_B_Redl_fits, where the actual calculation is performed, it might be more intuitive to rename j_dot_B_Redl_fits->j_dot_B_Redl and j_dot_B_Redl->j_dot_B_Redl_geom